### PR TITLE
Milestone: Added calculator component :1234:  

### PR DESCRIPTION
**Pull request by ahmedeid6842**
# 🚀 Level Up  Math Magicians with the Calculator Component: Crunch Numbers Like a Pro! 🧮

### Done: :heavy_check_mark: :memo: 
 - [x] Delete all the boilerplate from your Create React App, including text, images, and styles.
 - [x] Create a new directory called `components`.
 - [x] Inside the `components` directory, create a new file called `calculator.js`.
 - [x] Implement your Calculator component using class-based components for maximum flexibility and functionality.
 
4zFo

### CODE REVIEW

File: src/components/calculator.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import { Component } from 'react';
+import {
+  Row, Col, Button, Form,
+} from 'react-bootstrap';
+
+class Calculator extends Component {
```
Comment (Ismailco) [inline src/components/calculator.jsx:6]: - [ ] The `Calculator.jsx` component should be a functional component, kindly make sure to change your code so this component is a functional component

Comment (Ismailco) [inline src/components/calculator.jsx:6]: - [ ] The `Calculator.jsx` should have at least 1 child component, kindly make sure to split the code and add a child component to this one

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Component } from 'react';
+import Calculator from './calculator';
+
+class App extends Component {
```
Comment (Ismailco) [inline src/components/App.js:4]: - [ ] [OPTIONAL] Kindly make sure the app component is a functional component and not a class-based component
- [ ] [OPTIONAL] I suggest moving the `App.js` into the `src` folder outside of the `components` directory, this is the standard file structure for React

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @ahmedeid6842, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.jsx (hunk 1 above)
Comment (ahmedeid6842) [reply src/components/calculator.jsx:6]: @Ismailco 
I created a child component called `Calculator` which fulfills the activity requirement of **"Create at least 1 child component"**. I then used this component in `App.js`.

Thank you for reviewing my work. While I appreciate your input :pray: , I respectfully disagree about the required change, as I have already completed it and believe it meets the requirements for this part.

Comment (ahmedeid6842) [reply src/components/calculator.jsx:6]: @Ismailco 
Thank you for your response. I appreciate your help :+1: :pray: . However, I would like to clarify that the project requirements do not specify the location of the child component, only that it should be created.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @ahmedeid6842 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmedeid6842** [comment]
@Ismailco , @emmanuellekamwa 

Thanks guys for your time and efforts to review my code and provide feedback :pray: :handshake:

**Comment by ahmedeid6842** [comment]
# 🎉🎉🥳🥳

7fb4d55fa09fc4f5426e6d5934d1d02f

**Pull request merged**